examples/simulation-pytorch/sim.py

```diff
@@ -78,18 +80,33 @@ def evaluate(self, parameters, config):
         return float(loss), len(valloader.dataset), {"accuracy": float(accuracy)}
 
 
-def get_client_fn(train_partitions, val_partitions):
+def get_client_fn(dataset: FederatedDataset):
     """Return a function to construct a client.
 
-    The VirtualClientEngine will exectue this function whenever a client is sampled by
+    The VirtualClientEngine will execue this function whenever a client is sampled by
     the strategy to participate.
     """
 
     def client_fn(cid: str) -> fl.client.Client:
         """Construct a FlowerClient with its own dataset partition."""
 
-        # Extract partition for client with id = cid
-        trainset, valset = train_partitions[int(cid)], val_partitions[int(cid)]
+        # Let's get the partition corresponding to the i-th client
+        client_dataset = dataset.load_partition(int(cid), "train")
+
+        # Now let's split it into train (90%) and validation (10%)
+        client_dataset_splits = client_dataset.train_test_split(test_size=0.1)
+
+        trainset = client_dataset_splits["train"]
+        valset = client_dataset_splits["test"]
+
+        # Now we apply the transform to each batch.
+        trainset = trainset.map(
+            lambda img: {"img": get_mnist_transforms()(img)}, input_columns="image"
+        ).with_format("torch")
+
+        valset = valset.map(
+            lambda img: {"img": get_mnist_transforms()(img)}, input_columns="image"
+        ).with_format("torch")
```

Comment:
I think it's better to keep the names the same. Since the dataset has "image" we should return "image". The current approach produces a bigger dataset with redundant data (the values from the "image" stay the same and we add new "column" of the name "img"). It's better to simply overwrite the values not to store unnecessary data. I clarified it also in the how-to-guide re PyTorch